rust_snuba/src/strategies/clickhouse/writer_v2.rs

```diff
@@ -0,0 +1,215 @@
+use std::sync::Arc;
+use std::time::{Duration, SystemTime};
+
+use reqwest::header::{HeaderMap, HeaderValue, ACCEPT_ENCODING, CONNECTION};
+use reqwest::{Client, Response};
+use sentry_arroyo::processing::strategies::run_task_in_threads::{
+    ConcurrencyConfig, RunTaskError, RunTaskFunc, RunTaskInThreads, TaskRunner,
+};
+use sentry_arroyo::processing::strategies::{
+    CommitRequest, ProcessingStrategy, StrategyError, SubmitError,
+};
+use sentry_arroyo::types::Message;
+use sentry_arroyo::{counter, timer};
+
+use crate::config::ClickhouseConfig;
+use crate::types::{BytesInsertBatch, RowData};
+
+const CLICKHOUSE_HTTP_CHUNK_SIZE: usize = 1_000_000;
+
+fn clickhouse_task_runner(
+    client: Arc<ClickhouseClient>,
+    skip_write: bool,
+) -> impl TaskRunner<BytesInsertBatch<RowData>, BytesInsertBatch<()>, anyhow::Error> {
+    move |message: Message<BytesInsertBatch<RowData>>| -> RunTaskFunc<BytesInsertBatch<()>, anyhow::Error> {
+        let skip_write = skip_write;
+        let client = client.clone();
+
+        Box::pin(async move {
+            let (empty_message, insert_batch) = message.take();
+            let batch_len = insert_batch.len();
+            let (rows, empty_batch) = insert_batch.take();
+            let encoded_rows = rows.into_encoded_rows();
+
+            let write_start = SystemTime::now();
+
+            // we can receive empty batches since we configure Reduce to flush empty batches, in
+            // order to still be able to commit. in that case we want to skip the I/O to clickhouse
+            // though.
+            if encoded_rows.is_empty() {
+                tracing::debug!(
+                    "skipping write of empty payload ({} rows)",
+                    batch_len
+                );
+            } else if skip_write {
+                tracing::info!("skipping write of {} rows", batch_len);
+            } else {
+                tracing::debug!("performing write");
+
+                let response = client
+                    .send(encoded_rows.as_slice())
+                    .await
+                    .map_err(RunTaskError::Other)?;
+
+                tracing::debug!(?response);
```

Comment:
This is logging the entire response (might pollute the logs). Consider reducing the logging to what's needed.